invenio_app_ils/ui/src/invenio_app_ils/authentication/services/AuthenticationService.js

```diff
@@ -0,0 +1,31 @@
+class AuthenticationService {
+  getTokenFromDOM = () => {
+    const element = document.getElementsByName('authorized_token');
+    if (element.length > 0 && element[0].hasAttribute('value')) {
+      return element[0].value;
+    }
+    return '';
+  };
+
+  decodeProductionToken = token => {
+    if (process.env.NODE_ENV !== 'production') {
+      throw new Error('Token decoding is available only on production mode.');
+    }
+    if (!token) {
+      throw new Error('Token is empty!');
+    }
+    return JSON.parse(atob(token.split('.')[1]));
+  };
+
+  login = nextUrl => {
+    window.location =
```

Comment:
Not sure this is the react way. I might be wrong, but I would try something with `withRouter` and `history.push(myUrl)` instead of reassigning the global object.

Reply:
Is needed because we need to redirect outside react. React-router is handling urls internally so it wouldn't work in our case as we need to redirect to a flask url.